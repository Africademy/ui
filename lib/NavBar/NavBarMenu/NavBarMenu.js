"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Icon = require("../../Icon");

var _style = require("./style");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** NavBar Menu component used by the Select component to show a custom User name and avatar
 *  button */
var NavBarMenu = function NavBarMenu(_ref) {
  var user = _ref.user,
      onClick = _ref.onClick;
  var name = user.name,
      email = user.email,
      avatar = user.avatar,
      menuItems = user.menuItems;
  return _react.default.createElement(_style.NavBarStyled, null, _react.default.createElement(_style.NavBarUser, null, name && _react.default.createElement(_style.NavBarName, null, name), _react.default.createElement(_style.NavBarEmail, null, email)), avatar && _react.default.createElement(_style.NavBarAvatar, {
    avatar: user.avatar,
    onClick: onClick
  }), menuItems.length > 0 && _react.default.createElement(_style.NavBarChavron, {
    onClick: onClick
  }, _react.default.createElement(_Icon.ChevronDown, {
    color: "grayLight",
    size: "large"
  })));
};

NavBarMenu.propTypes = {
  /** User Name and Email to be shown in the NavBar */
  user: _propTypes.default.shape({
    name: _propTypes.default.string,
    email: _propTypes.default.string.isRequired,
    avatar: _propTypes.default.string,
    menuItems: _propTypes.default.array
  }).isRequired,

  /** OnClick function to be called on Avatar click, passed by the Select component */
  onClick: _propTypes.default.func.isRequired
};
var _default = NavBarMenu;
exports.default = _default;