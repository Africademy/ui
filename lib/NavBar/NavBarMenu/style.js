"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NavBarChavron = exports.NavBarEmail = exports.NavBarName = exports.NavBarAvatar = exports.NavBarUser = exports.NavBarStyled = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _colors = require("../../style/colors");

var _fonts = require("../../style/fonts");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  background: #3D3D3D;\n  border: 1px solid #3D3D3D;\n  box-sizing: border-box;\n  border-radius: 4px;\n  color: ", ";\n  height: 16px;\n  width: 16px;\n  display: flex;\n  align-items: center;\n  position: relative;\n  left: -8px;\n  cursor: pointer;\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n  font-size: ", ";\n  color: ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n  font-size: 16px;\n  color: ", ";\n  font-weight: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  border-radius: 100px;\n  width: 40px;\n  height: 40px;\n  background: ", ";\n  cursor: pointer;\n  background-size: cover;\n  background-image: url(", ");\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  align-items: end;\n  flex-direction: column;\n  padding-right: 20px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  background: ", ";\n  padding: 0px 8px 0px 16px;\n  height: 64px;\n  width: 230px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var NavBarStyled = _styledComponents.default.div(_templateObject(), _colors.grayDark);

exports.NavBarStyled = NavBarStyled;

var NavBarUser = _styledComponents.default.div(_templateObject2());

exports.NavBarUser = NavBarUser;

var NavBarAvatar = _styledComponents.default.div(_templateObject3(), _colors.gray, function (props) {
  return props.avatar;
});

exports.NavBarAvatar = NavBarAvatar;

var NavBarName = _styledComponents.default.div(_templateObject4(), _colors.white, _fonts.fontWeightMedium);

exports.NavBarName = NavBarName;

var NavBarEmail = _styledComponents.default.div(_templateObject5(), _fonts.fontSize, _colors.grayLight);

exports.NavBarEmail = NavBarEmail;

var NavBarChavron = _styledComponents.default.div(_templateObject6(), _colors.gray);

exports.NavBarChavron = NavBarChavron;