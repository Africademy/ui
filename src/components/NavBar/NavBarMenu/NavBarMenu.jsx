import React from 'react';
import PropTypes from 'prop-types';
import { ChevronDown } from '../../Icon';
import {
  NavBarStyled,
  NavBarEmail,
  NavBarName,
  NavBarUser,
  NavBarAvatar,
  NavBarChavron,
  NavBarImpersonating,
} from './style';

export function getUserAvatar(user) {
  if (user.avatar) {
    return user.avatar;
  }

  return 'https://s3.amazonaws.com/buffer-ui/Default+Avatar.png';
}

/** NavBar Menu component used by the Select component to show a custom User name and avatar
 *  button */
const NavBarMenu = ({ user, onClick, isImpersonation }) => (
  <NavBarStyled onClick={onClick} isImpersonation={isImpersonation}>
    <NavBarUser>
      {isImpersonation && (
        <NavBarImpersonating user={user}>Impersonating</NavBarImpersonating>
      )}
      <NavBarName isImpersonation={isImpersonation}>{user.name}</NavBarName>
      <NavBarEmail isImpersonation={isImpersonation}>{user.email}</NavBarEmail>
    </NavBarUser>
    <NavBarAvatar avatar={getUserAvatar(user)} onClick={onClick} />
    <NavBarChavron>
      <ChevronDown color="grayLight" size="large" />
    </NavBarChavron>
  </NavBarStyled>
);

NavBarMenu.propTypes = {
  /** User Name and Email to be shown in the NavBar */
  user: PropTypes.shape({
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,

  /** OnClick function to be called on Avatar click, passed by the Select component */
  onClick: PropTypes.func.isRequired,
  isImpersonation: PropTypes.bool,
};

NavBarMenu.defaultProps = {
  isImpersonation: false
};

export default NavBarMenu;
