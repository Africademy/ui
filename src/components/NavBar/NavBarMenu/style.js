import styled from 'styled-components';
import {
  gray, grayDark, grayLight, white,
} from '../../style/colors';
import { fontSize, fontWeightMedium } from '../../style/fonts';

export const NavBarStyled = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${grayDark};
  padding: 0px 16px;
  height: 64px;
  width: 230px;
`;

export const NavBarUser = styled.div`
  display: flex;
  align-items: end;
  flex-direction: column;
  overflow: hidden;
  margin-right: 20px;
`;

export const NavBarAvatar = styled.div`
  border-radius: 100px;
  width: 60px;
  height: 40px;
  background: ${gray};
  cursor: pointer;
  background-size: cover;
  background-image: url(${props => (props.avatar)});
`;

export const NavBarName = styled.div`
  font-size: 16px;
  color: ${white};
  font-weight: ${fontWeightMedium};
  width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export const NavBarEmail = styled.div`
  font-size: ${fontSize};
  color: ${grayLight};
  width: 100%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export const NavBarChavron = styled.div`
  background: #3D3D3D;
  border: 1px solid #3D3D3D;
  box-sizing: border-box;
  border-radius: 4px;
  color: ${gray};
  height: 16px;
  width: 16px;
  display: flex;
  align-items: center;
  position: relative;
  left: -8px;
  cursor: pointer;
`;
