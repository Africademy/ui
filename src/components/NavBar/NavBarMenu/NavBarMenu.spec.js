import snap from 'jest-auto-snapshots';
import md5 from 'blueimp-md5';
import NavBarMenu, { getUserAvatar } from './NavBarMenu';
import 'jest-styled-components';

describe('Avatar', () => {
  it('should return existing avatar URL', () => {
    expect(getUserAvatar({ avatar: 'foo'})).toBe(`foo`);
  });
  it('should return gravatar', () => {
    const user = { email: 'foo@bar.com' };
    expect(getUserAvatar(user)).toBe(`https://www.gravatar.com/avatar/${md5(user.email)}?d=https%3A%2F%2Fs3.amazonaws.com%2Fbuffer-ui%2FDefault%2BAvatar.png`);
  });
});

describe('NavBarMenu methods', () => {
});


snap(NavBarMenu, './NavBarMenu.jsx');
