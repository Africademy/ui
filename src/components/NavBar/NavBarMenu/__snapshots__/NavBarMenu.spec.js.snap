// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > NavBarMenu Matches snapshot when passed only required props 1`] = `
<ForwardRef(styled.div)
  onClick={[MockFunction]}
>
  <ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      jest-auto-snapshots String Fixture
    </ForwardRef(styled.div)>
    <ForwardRef(styled.div)>
      jest-auto-snapshots String Fixture
    </ForwardRef(styled.div)>
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)
    avatar="https://s3.amazonaws.com/buffer-ui/Default+Avatar.png"
    onClick={[MockFunction]}
  />
  <ForwardRef(styled.div)>
    <ChevronDownIcon
      color="grayLight"
      size="large"
    />
  </ForwardRef(styled.div)>
</ForwardRef(styled.div)>
`;
